src/chart/generateCategoricalChart.js

```diff
@@ -44,6 +44,7 @@ const generateCategoricalChart = (ChartComponent, GraphicalChild) => {
     static propTypes = {
       ...ChartComponent.propTypes,
       syncId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+      compact: PropTypes.bool,
```

Comment:
@xile611 For this `compact` prop, my main consideration was to exclude:

- HTML DOMs including <div>, Legend and Tooltip
- X-axises and y-axises
- Brush

All those exclusions are specifically made for the "panoramic brush" use case. Do you think this is way too specific? When looking back into this, I found myself may still need grid inside the panorama. Maybe it's better to extend this prop to an array (of chart parts to be excluded, like `['grid', 'chart', 'children']`... a little tricky...)?

Anyway for this MR, this prop is necessary, since when cloning element in `Brush`:  https://github.com/recharts/recharts/pull/586/files/6c29a98d8e445ffe456d968cd9b0a3080e47b5bb#diff-44d40453cfbdcbd566bcc8d82acd5c5bR306 , I don't have any chance to exclude parts inside the wrapping `Surface`.

Looking forward to your further feedbacks.

----

About https://github.com/recharts/recharts/issues/205 you mentioned, since certain parts (HTML, Tooltip, etc.) would be excluded by setting `compact`, it would naturally become one single exportable svg. But I don't think this meets the requirement in #205, as Legend is excluded as well. 

Just like you mentioned in that ticket, (HTML) DOM nodes handles flow layout better than SVG nodes. My experience also tells that embedding HTML DOM node in SVG node is much harder than embedding SVG node in HTML DOM ones, due to poor compatibility of svg foreignObject: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject#Browser_compatibility. So any component that contains, or possibly contains HTML, is better to be rendered at outside of chart's main SVG, which is exactly the current solution in Recharts (i.e. Legend and Tooltip).

As #205 requested, exporting to all-in-one SVG, requires either: 

- Implement Legend with pure SVG, need to calculate text's width and handle potential line wrapping, or:
- Keep using HTML, but convert it to SVG after rendered, unfortunately this is not doable in browser: http://stackoverflow.com/a/26963688/245345 , so this must be done together with a backend.